Adding documentation related to raven 5 persistence for Audit (#5887)
Particular/docs.particular.net: servicecontrol/how-purge-expired-data.md

related:
 - nservicebus/recoverability
 - nservicebus/operations/auditing
 - servicecontrol/audit-instances/persistence
reviewed: 2022-10-18
---

ServiceControl stores audit and error data. Any audit and error data that is older than the specified thresholds is deleted from the embedded RavenDB. The expiration thresholds for both faulted and audited messages must be set during installation. These values can be modified later by launching ServiceControl Management and editing the configuration settings for the instance.

Note: The expiration process curates only the data in the embedded RavenDB. Audit and error forwarding queues are not curated or managed by ServiceControl. To turn these settings off, launch ServiceControl Management and edit the configuration settings for the instance.

Warning: The database will not automatically shrink in size after reducing the retention period. Ensure ServiceControl had time to purge all expired messages and then [Compact the database](db-compaction.md).

> ## Differences in purge implementations

Reviewer: ```suggestion
## Differences in message rentention implementations
```

> Expiration of error and audit data is implemented by a reccuring task that checks the documents for old data and deletes documents that have expired. Changing the expiration setting results in it being picked up during the next time the service is run. 

Reviewer 2: ```suggestion
The expiration of error and audit data is implemented by a recurring task that checks and deletes expired documents. Changing the expiration setting results in it being picked up the next time the service is run. 
```

> Expiration of error and audit data is implemented by runnig a reccuring task that checks the documents for old data and deleting the rows that should be deleted. Changing the expiration setting results in it being picked up during the next time the service is run. 

Reviewer 3: ```suggestion
Expiration of error and audit data is implemented by a reccuring task that checks the documents for old data and deletes documents that have expired. Changing the expiration setting results in it being picked up during the next time the service is run. 
```

> In the ServiceControl.Audit version 4.26 and up expiration is done using the persistance mechanisms, that results in metadata key to be stored with every message. Thanks to that, purging of the data can be handled fully by database itself. The drawback is that changing of expiration setting is only applied to new rows. 

Reviewer 3: Using version language guidance.

```suggestion
In ServiceControl.Audit version 4.26 and above expiration is handled by the database automatically. Each audited message contains a metadata key that tells the database when to remove it. When the expiration setting changes, the new value is only applied to new audit data. Existing audit meta is not changed. Messages already in the database are not affected by the change to the expiration setting.
```

> In ServiceControl.Audit version 4.26 and above expiration is handled by the database automatically. Each audited message contains a metadata key that tells the database when to remove it. When the expiration setting changes, the new value is only applied to new audit data. Existing audit meta is not changed. Messages already in the database are not affected by the change to the expiration setting.

Reviewer 2: ```suggestion
In ServiceControl.Audit version 4.26 and above expiration is handled by the database automatically. Each audited message contains a metadata key that tells the database when to remove it. When the expiration setting changes, the new value is only applied to new audit data. Metadata for already ingested messages is not changed. Only new audited messages are affected by the change to the expiration setting.
```
